include/libtorrent/alert_types.hpp

```diff
@@ -2382,12 +2382,39 @@ namespace libtorrent
 		int m_peers_idx;
 	};
 
+	// This is posted exactly once for every call to session_handle::dht_direct_request.
+	// If the request failed, response() will return a default constructed bdecode_node.
+	struct TORRENT_EXPORT dht_direct_response_alert: alert
+	{
+		dht_direct_response_alert(aux::stack_allocator& alloc, void* userdata
+			, udp::endpoint const& addr, bdecode_node const& response);
+
+		// for when there was a timeout so we don't have a response
+		dht_direct_response_alert(aux::stack_allocator& alloc, void* userdata
+			, udp::endpoint const& addr);
+
+		TORRENT_DEFINE_ALERT(dht_direct_response_alert, 88)
+
+		static const int static_category = alert::dht_notification;
```

Comment:
in some ways that would make sense, but on the other hand, you only get this if you explicitly make a request, so this won't cause any spam really. and the risk with having this be filtered is that one may be confused why there is no response. So, it probably makes sense to keep it in this category. It may even make sense to post this alert unconditionally, and perhaps even make it non-discardable.